packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts

```diff
@@ -763,6 +763,9 @@ export async function pollTransformationJob(jobId: string, validStates: string[]
                     transformByQState.setJobFailureErrorChatMessage(
                         CodeWhispererConstants.failedToStartJobLinesLimitChatMessage
                     )
+                }
+                if (errorMessage.includes('The requested module could not be built')) {
+                    throw new TransformationPreBuildError(response.transformationJob.reason || '')
```

Comment:
Ideally we don't want to rely on comparing strings, since this is a one-way door decision that will break if service side changes the error message, or constrain service side from updating the string. Instead, as a short term mitigation, we can attempt to pre-fetch the artifact for all PreBuild failures, then only show the UI (i.e. throw TransformationPreBuildError) if artifact was available from the download attempt.